libs/media-player/src/lib/timeline.tsx

```diff
@@ -0,0 +1,501 @@
+import { isNullOrUndefined } from '@coscrad/validation-constraints';
+import { ZoomIn as ZoomInIcon, ZoomOut as ZoomOutIcon } from '@mui/icons-material';
+import { Box, Divider, IconButton, Tooltip, styled } from '@mui/material';
+import { RefObject, useEffect, useRef, useState } from 'react';
+import './interactive-media-player-style.css';
+import { RangeBar } from './range-bar';
+import { asFormattedMediaTimecodeString } from './shared/as-formatted-media-timecode-string';
+import { EDITOR_SOUND_BAR_HEIGHT_IN_PIXELS, ZOOM_LEVELS, ZoomLevels } from './shared/constants';
+import { convertTimecodeToTimelineUnits } from './shared/convert-timecode-to-timeline-units';
+import { convertTimelineUnitsToTimecode } from './shared/convert-timeline-units-to-timecode';
+import { getZoomLevelConfig } from './shared/get-zoom-level';
+import { TimelineRuler } from './timeline-ruler';
+
+const WAVE_FORM_URL = 'https://guujaaw.info/images/audio-wave-form.png';
```

Comment:
This is a bit awkward. I wonder if we should put this in the content config for lack of a better place.